hr_timesheet_sheet/models/account_analytic_line.py

```diff
@@ -8,62 +8,37 @@
 class AccountAnalyticLine(models.Model):
     _inherit = 'account.analytic.line'
 
-    sheet_id_computed = fields.Many2one(
-        comodel_name='hr_timesheet.sheet',
-        string='Sheet',
-        compute='_compute_sheet',
-        index=True,
-        ondelete='cascade',
-        search='_search_sheet'
-    )
     sheet_id = fields.Many2one(
         comodel_name='hr_timesheet.sheet',
         string='Sheet',
-        compute='_compute_sheet',
-        store=True,
     )
 
-    @api.depends('date', 'user_id', 'project_id', 'task_id', 'company_id',
-                 'sheet_id.date_start', 'sheet_id.date_end',
-                 'sheet_id.employee_id', 'sheet_id.company_id')
     def _compute_sheet(self):
         """Links the timesheet line to the corresponding sheet"""
         for timesheet in self:
-            if timesheet.sheet_id or not timesheet.project_id:
+            if not timesheet.project_id:
                 continue
-            sheets = self.env['hr_timesheet.sheet'].search(
-                [('date_end', '>=', timesheet.date),
-                 ('date_start', '<=', timesheet.date),
-                 ('employee_id.user_id.id', '=', timesheet.user_id.id),
-                 ('company_id', 'in', [timesheet.company_id.id, False]),
-                 ('state', '=', 'draft'),
-                 ])
-            if sheets:
-                timesheet.sheet_id_computed = sheets[0]
-                timesheet.sheet_id = sheets[0]
+            sheet = self.env['hr_timesheet.sheet'].search([
+                ('date_end', '>=', timesheet.date),
+                ('date_start', '<=', timesheet.date),
+                ('employee_id', '=', timesheet.employee_id.id),
+                ('company_id', 'in', [timesheet.company_id.id, False]),
+                ('state', '=', 'draft'),
+            ], limit=1)
+            timesheet.sheet_id = sheet
 
-    def _search_sheet(self, operator, value):
-        assert operator == 'in'
-        ids = []
-        for ts in self.env['hr_timesheet.sheet'].browse(value):
-            self._cr.execute("""
-                    SELECT l.id
-                        FROM account_analytic_line l
-                    WHERE %(date_end)s >= l.date
-                        AND %(date_start)s <= l.date
-                        AND %(user_id)s = l.user_id
-                        AND %(company_id)s = l.company_id
-                    GROUP BY l.id""", {'date_start': ts.date_start,
-                                       'date_end': ts.date_end,
-                                       'user_id': ts.employee_id.user_id.id,
-                                       'company_id': ts.company_id.id,
-                                       })
-            ids.extend([row[0] for row in self._cr.fetchall()])
-        return [('id', 'in', ids)]
+    @api.model
+    def create(self, values):
+        res = super().create(values)
```

Comment:
`super(AccountAnalyticLine, self)` to make it compatible with python 2.7 just in case